tmt/beakerlib.py

```diff
@@ -100,20 +120,24 @@ def __init__(self, identifier, parent=None):
                 raise tmt.utils.SpecificationError(
                     f"Library name '{self.name}' does not start with a '/'.")
             # Use provided repository nick name or parse it from the url/path
-            self.repo = identifier.get('nick')
-            if not self.repo and self.url:
-                try:
-                    self.repo = re.search(
-                        r'/([^/]+?)(/|\.git)?$', self.url).group(1)
-                except AttributeError:
+            repo = identifier.get('nick')
+            if not repo and self.url:
+                repo_search = re.search(r'/([^/]+?)(/|\.git)?$', self.url)
+                if not repo_search:
                     raise tmt.utils.GeneralError(
                         f"Unable to parse repository name from '{self.url}'.")
-            if not self.repo and self.path:
+                repo = repo_search.group(1)
+            elif not repo and self.path:
                 try:
-                    self.repo = os.path.basename(self.path)
+                    repo = os.path.basename(self.path)
                 except TypeError:
                     raise tmt.utils.GeneralError(
                         f"Unable to parse repository name from '{self.path}'.")
+            else:
+                raise tmt.utils.GeneralError(
+                    f"Unable to parse repository name from '{self.path}'.")
```

Comment:
Why are you adding this extra `else` branch? This seems to break `/tests/discover/libraries/` as `repo` can also be detected from `nick` (alread set before the condition).

Reply:
hmm, I will need to check

Reply:
Indeed, thanks, updated. The thing is, `nick` can be `None`, so `mypy` complains if that can happen. I changed the check to check for that and return a (I hope) resonable error.